index.rst

```diff
@@ -46,14 +46,314 @@
+
+To understand this behavior we have to look at the how the DDS Communication works when operating in single process mode.
+As explained in :ref:`DDS-Communication-Architecture`, each component running in single process mode will also run a full set of DDS services.
+By design, these services will be processing not only the data required by the application itself but also in the full topic set present in the system at any moment.
+This explains the extremely high inter-connectivity observed in :numref:`fig-net-topology-full-community` and also why upcoming components would lag behind in trying to keep up with the system traffic.
+
+Therefore, following the `OpenSplice deployment guidelines`_ on how to deal with large scale systems, we proceeded to experiment with OpenSplice shared memory mode.
+
+.. _OpenSplice deployment guidelines: https://istkb.adlinktech.com/article/opensplice-dds-v6-offers-choice-two-memory-modes-shared-memory-single-process-i-use/
+
+.. _NCSA-Shared-Memory-Test:
+
+NCSA Shared Memory Test
+=======================
+
+The first step in testing shared memory mode was to rebuild all the components using the licensed version of the OpenSplice, since the community edition does not support this mode of operation.
+This required some refactoring of the build scripts since the licensed version is stored in a password-protected private repository, and we also want to be able to build both licensed and community editions.
+Furthermore, we also have to provide the ospl daemon to run the OpenSplice service.
+
+Since all the services run from a docker containers, the daemon was also deployed on a container.
+To make sure the system would work properly we had to enable two additional features when running the containers:
+
+  * ``ipc=host``, to allow containers to access the host machine shared memory space and,
+  * ``pid=host``, to allow containers to share the process id with the host machine.
+
+In addition the container must also share a common ``/tmp`` folder with read and write access to all users and groups.
+This is required so containers can access a socket and a lock file created by the ospl daemon and needed for the shared memory process to work.
+
+Finally, shared memory is activate by use of a specific :ref:`ospl configuration <Appendix-Shared-Memory-OpenSplice-Configuration>`, which must be the same for the daemons and applications.
+
+Once all the components where updated to the licensed version of OpenSplice and with the new configuration and the daemons where ready, they where deployed into a fresh system (see :ref:`NCSA-Standard-Deployment-Test` for details).
+The daemons myst be deployed first, so that the OpenSplice DDS services become available.
+Next we brought up all the CSCs, one by one, and salkafka producers following the same process of the :ref:`standard configuration test <NCSA-Standard-Deployment-Test>`.
```

Comment:
What happens if you try to bring everything up at once, like a more automated system would do?

Reply:
Should work normally... we choose to do it this way so we can monitor the network traffic at each stage... I'll mention that in the text.